### PR TITLE
Onboarding: new welcome page

### PR DESCRIPTION
**Pull request by PopDaph**
## Context

This task is part of the new onboarding for members joining a workspace. 
After landing on the `/join` page and going through Google Sign in, instead of landing on the Assistant, they will land on this new welcome page, where we ask the user to confirm their name and let us know about their AI expertise. 

## Tech choices

**Data Model**
I decided to store the expertise as a `UserMetadata` instead of introducing a new field on the `User` model. 

**Form Submission**
The form submission is allowed only when all fields are filled. 
When we submit:
- we first PATCH on `/api/user` -> this is a bit weird to patch without an id in the url but we get the user from the api route. 
- then if successful we post the metadata on the existing route for this model.
- if something is wrong among those 2 calls, we won't block the user and she will go to next step.

**Redirection**
Once the form is submitted, user will be redirected on the Assistant or directly to a conversation if they are onboarding in the flow "[domain_conversation_link](https://github.com/dust-tt/dust/blob/main/front/pages/w/%5BwId%5D/join.tsx#L10-L27)". 

## Screenshot - web 
Capture d’écran 2023-10-13 à 13 42 28

## Screenshot - mobile 

Capture d’écran 2023-10-13 à 13 42 06

## Next step

- Introduce the video page before redirecting to the Assistant.
- Cleaning: Introduce a proper OnboardingLayout that will be used for join / welcome and the video page.

### CODE REVIEW

File: front/lib/api/user.ts [hunk 1]
```diff
@@ -57,3 +58,30 @@ export async function setUserMetadata(
   metadata.value = update.value;
   await metadata.save();
 }
+
+export async function updateUserFullName({
+  userId,
+  firstName,
+  lastName,
+}: {
+  userId: ModelId;
```
[the remark this answers is not in the document]
Comment (spolu) [reply front/lib/api/user.ts:67]: Let's pass `user: UserType, firstName: string, lastName: string` ?

File: front/pages/w/[wId]/welcome.tsx [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+  const handleSubmit = async () => {
+    const updateUserFullNameRes = await fetch("/api/user", {
+      method: "PATCH",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({ firstName, lastName }),
+    });
+    if (updateUserFullNameRes.ok) {
+      await fetch("/api/user/metadata/expertise", {
```
Comment (spolu) [inline front/pages/w/[wId]/welcome.tsx:83]: UserMetadata are meant to be deletable at will without breaking the product (think cookie). Let's not use that for this if we need to use it at a later stage :+1:

**Review by spolu — COMMENTED** [review]
Left some comments otherwise LGTM.

Why not make every new user go through that?

**Comment by spolu** [comment]
OOC What's the purpose of the expertise question?

**Comment by spolu** [comment]
Purpose may mean that we can use UserMetadata indeed...

**Comment by PopDaph** [comment]
> Why not make every new user go through that?

A user onboarding from the homepage will also land on a welcome page but the form is a bit different, and to keep is simple I'm preparing PR for each. -> we will be able to reuse most of this code tho!

**Comment by spolu** [comment]
> > Why not make every new user go through that?
> 
> A user onboarding from the homepage will also land on a welcome page but the form is a bit different, and to keep is simple I'm preparing PR for each. -> we will be able to reuse most of this code tho!

OK! :+1:

**Comment by PopDaph** [comment]
For now we don't do anything with the expertise question, but Ed wanted it. Until we know what we want to do with it I'd rather keep it here, even if that means having to run a migration someday. WDYT?

**Comment by PopDaph** [comment]
@spolu copy updated and I introduced an Onboarding layout shared between /join and /welcome, with the background animation. 
Capture d’écran 2023-10-13 à 15 03 42

PTAL? 🙏🏻

**Review by spolu — APPROVED** [review]

**Pull request merged**